app/models/educator.rb

```diff
@@ -76,6 +80,16 @@ def is_authorized_for_school(currentSchool)
     (self.has_access_to_grade_levels? && self.school == currentSchool)
   end
 
+  def is_authorized_for_section(section)
+    return true if self.districtwide_access?
+
+    return false if self.school.present? && self.school != section.course.school
+
+    return true if self.schoolwide_access? || self.admin?
+    return true if section.in?(self.sections)
+    false
+  end
+
```

Comment:
Districtwide can see all sections
Schoolwide can see all sections in their school
Otherwise, educators can see sections they are assigned to.

We may want to revisit the logic that limits educators to only seeing students in the educator's assigned school unless they are district-wide. I know there are elementary specialists (library and music) who teach at multiple elementary schools.